merlin-server/sql/merlin/tables/activity.sql

```diff
@@ -83,3 +86,59 @@ create trigger increment_revision_on_delete_activity_trigger
 after delete on activity
 for each row
 execute function increment_revision_on_delete_activity();
+
+create function check_activity_directive_metadata()
```

Comment:
This is really awesome - I'm impressed both by postgres's json capabilities, and how quickly you've learned to use them.

Just to confirm a scenario in my head: this validation happens on every insert or update.
1. A key in the schema is changed to have a different type
2. An existing activity's start time is updated (e.g. by dragging in the UI)
3. This function will run, and reject the update (essentially "freezing" all updates to that activity until the metadata is fixed)

Now, perhaps the answer is "don't change the schema for existing keys, just make new keys and delete old ones" - I just want to confirm that that's the desired behavior.

Reply:
I think the appropriate handling of that scenario is to clear out the metadata field in a second submission from the client to update - try-fail-retry, assuming it isn't already setting the value with the "new" schema.

Reply:
I wonder if we can just do the validation when the specific metadata field changes...